Doc/whatsnew/3.7.rst

```diff
@@ -330,6 +330,11 @@ that may require changes to your code.
 Changes in the Python API
 -------------------------
 
+* :meth:`pkgutil.walk_packages` now raises ValueError/TypeError if path is
+  string/bytes object.
+  Previously an empty list was returned. (Contributed by Sanyam Khurana in
+  :issue:`24744`.)
```

Comment:
Please reflow this into a single paragraph.  And it should only mention str; it has always raised a TypeError for bytes.